pkg/karmadactl/unjoin.go

```diff
@@ -16,12 +16,21 @@ import (
 	"k8s.io/client-go/rest"
 	"k8s.io/klog/v2"
 
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
 	karmadaclientset "github.com/karmada-io/karmada/pkg/generated/clientset/versioned"
 	"github.com/karmada-io/karmada/pkg/karmadactl/options"
 	"github.com/karmada-io/karmada/pkg/util"
 	"github.com/karmada-io/karmada/pkg/util/names"
 )
 
+const (
+	agentClusterRole        = "karmada-agent"
+	agentClusterRoleBinding = "karmada-agent"
+	agentDeployment         = "karmada-agent"
+	agentNamespace          = "karmada-system"
+	agentServiceAccount     = "karmada-agent-sa"
```

Comment:
As we talked at the meeting, we can assume that the `Deployment` name is `karmada-agent`.
But I just realized what we have to clean up is more than that.

I'm getting hesitant because I know some users install the things at another namespace(not `karmada-system`).
So, the solution might be not working properly in some scenarios.

Reply:
Let's see if other better solutions.

Reply:
Thanks for your reviews, @RainbowMango 
Yes, it is a problem. Could we record the agent information(namespace/name) in `annotation` or `Status` of the `Cluster  crd` when we join the agent? Then we can delete the agent through the agent information.